vital/tests/test_detected_object_set.cxx

```diff
@@ -108,39 +108,41 @@ IMPLEMENT_TEST(object_creation)
   // get whole list sorted by confidence
   auto list_1 = do_set.select();
 
-  TEST_EQUAL( "size of list", list_1.size(), 5 );
-
-  TEST_EQUAL( "expected confidence 0", list_1[0]->confidence(), 1.0 );
-  TEST_EQUAL( "expected confidence 1", list_1[1]->confidence(), 0.78 );
-  TEST_EQUAL( "expected confidence 2", list_1[2]->confidence(), 0.75 );
-  TEST_EQUAL( "expected confidence 3", list_1[3]->confidence(), 0.70 );
-  TEST_EQUAL( "expected confidence 4", list_1[4]->confidence(), 0.65 );
+  TEST_EQUAL( "size of list", list_1->size(), 5 );
+  auto it = list_1->cbegin();
+  TEST_EQUAL( "expected confidence 0", (*it++)->confidence(), 1.0 );
+  TEST_EQUAL( "expected confidence 1", (*it++)->confidence(), 0.78 );
+  TEST_EQUAL( "expected confidence 2", (*it++)->confidence(), 0.75 );
+  TEST_EQUAL( "expected confidence 3", (*it++)->confidence(), 0.70 );
+  TEST_EQUAL( "expected confidence 4", (*it++)->confidence(), 0.65 );
```

Comment:
I agree that the double dereference is not as nice as it could be, but if you are writing a object filter and wanted to create another set with only selected objects, you need the shared-pointer so the object can be correctly managed.